persistence-modules/hibernate-mapping/src/main/java/com/baeldung/hibernate/wherejointable/Group.java

```diff
@@ -0,0 +1,85 @@
+package com.baeldung.hibernate.wherejointable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+
+@Entity(name = "e_group")
+public class Group {
+
+    @Id
+    @GeneratedValue
+    private Long id;
+
+    private String name;
+
+    @ManyToMany(mappedBy = "groups")
+    private List<User> users = new ArrayList<>();
+
+    public Group() {
+
+    }
+
+    public Group(String name) {
+        this.name = name;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public List<User> getUsers() {
+        return users;
+    }
+
+    public void setUsers(List<User> users) {
+        this.users = users;
+    }
+
+    @Override
+    public String toString() {
+        return "Group [name=" + name + "]";
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((name == null) ? 0 : name.hashCode());
+        return result;
+    }
```

Comment:
This is a rather messy hashcode... Do we need hashcode and equals for this implementation? If so, please generate using the more modern Java 7+ template.

Reply:
Although removing equals and hashCode doesn't break the tests here, it is a good practice to implement them, to define the equality of entities independent from synthetic identifiers. It might cause unexpected behaviors for example when we deal with collections like `.contains()` or adding an element to `Set` structures, etc.

So, I propose refactoring them to Java 7+ way and do not remove. What do you think?

Reply:
I think the issue here is if they are not needed by anything in our example code, then they're just noise. There's also a decision to justify which is why the hashcode does not hash all properties.

My general rule is that if it's hard to explain, try removing it ;)

Reply:
Sure, I removed them because we don't explain their effects here. I am more of a developer rather than an educator so, you have the point. But, I am getting closer I hope :)